client/modules/wj-util/src/index.ts

```diff
@@ -430,3 +434,42 @@ export function html(strings: TemplateStringsArray, ...subs: (string | string[])
 export function mod(a: number, n: number) {
   return ((a % n) + n) % n
 }
+
+/**
+ * Replaces a range inside of a string with a substitute.
+ *
+ * @param str - The string which should have a range inside of it replaced.
+ * @param from - The start of the replacement range.
+ * @param to - The end of the replacement range.
+ * @param sub - The replacement/substitute string.
+ */
+export function replaceRange(str: string, from: number, to: number, sub: string) {
+  return str.substr(0, from) + sub + str.substr(to)
+}
+
+/**
+ * Capitalizes (or lowercases, if set) a string. Does not uppercase the
+ * entire string, just the first character.
+ *
+ * @param str - The string to capitalize.
+ * @param state - Whether to capitalize or lowercase. Defaults to capitalize.
+ */
+export function capitalize(str: string, state = true) {
+  if (!state) return str.toLowerCase()
+  return replaceRange(str.toLowerCase(), 0, 1, str[0].toUpperCase())
+}
```

Comment:
Python uses "title" for this operation, to avoid confusion. When I first saw this I thought it meant upper-case.

Reply:
good idea